pkg/controller/pod.go

```diff
@@ -334,13 +344,35 @@ func (pnc *PodNetworksController) handleResult(
 			updatedStatus,
 		); setNetworkStatusError != nil {
 			klog.Errorf("error updating pod %s network status: %v", namespacedPodNameString, setNetworkStatusError)
+			shouldDeleteattachmentsAdded = true
+		}
+	}
+
+	var attachmentsToCancal []nadv1.NetworkSelectionElement
+	if shouldDeleteattachmentsAdded {
+		attachmentsToCancal = attachmentsToAdd
+	} else if addfailedIndex != -1 {
+		attachmentsToCancal = attachmentsToAdd[addfailedIndex:]
```

Comment:
```suggestion
		attachmentsToCancal = attachmentsToAdd[:addfailedIndex]
```
Shouldn't we select the n first elements instead of the n last ones? The results are reported in the same order as the in the annotations, if it fails at attachment 3, the 3 first attachments should be removed, not the 3 last ones.

Reply:
I think I misunderstood this part last time. Why do we need to delete the latest elements and not only the one at addfailedIndex? The controller haven't tried to ADD them, it stopped at addfailedIndex right?